packages/uicore/src/components/DropDown/DropDown.tsx

```diff
@@ -43,7 +43,12 @@ export interface DropDownProps
   iconProps?: Partial<IconProps>
   addClearBtn?: boolean
   placeholder?: string
-  getCustomLabel?: (item: SelectOption) => string | React.ReactElement
+  getCustomLabel?: (item: SelectOption) => string | ReactElement
+  /**
+   * Delays calling the items promise function.
+   * Instead of fetching items on mount, fetch whenever dropdown is opened OR there exists a selected item
+   */
+  isLazyItemsQuery?: boolean
```

Comment:
`isLazyItemsQuery` can be just `lazy` or `lazyLoadItems`

Reply:
Initially went with the same but not to be confused with the lazy loading of large number of dropdown items. Will update it to lazyItems